roles/pul_nomad/README.md

```diff
@@ -1,36 +1,220 @@
+
+Every Nomad Client & Server install gets `/etc/nomad.d/base.hcl` created from [`templates/nomad/base.hcl.j2`](templates/nomad/base.hcl.j2).
+
+Every Server install gets `/etc/nomad.d/server.hcl` created from [`templates/nomad/server.hcl.j2`](templates/nomad/server.hcl.j2).
+
+Every Client install gets `/etc/nomad.d/client.hcl` created from [`templates/nomad/client.hcl.j2`](templates/nomad/client.hcl.j2)
+
+The "gossip encryption key" is a key used to encrypt communication between Nomad server installations. We use the same key for both Consul & Nomad.
+
+### Consul Interaction
+
+Nomad uses Consul for clustering (the Nomad server installs find each other by querying
+Consul) and service discovery (when you launch a job in Nomad it can register a
+service in Consul that we can point to from our load balancers.)
+
+### Host Volumes
+
+Host volumes are a way for containers to save data to the underlying client. It's a folder on the client which the container maps - if a job requires a certain host volume it will only ever get scheduled on a node which has that volume. Most containers have no need for persistent data on disk - they store that data in a database or something similar, but if we run a database or something like that in Nomad it might need a host volume. In those cases, we must ensure high availability through some other means (e.g "cloud" mode on Solr, clusters in Postgres, HA Mode in Redis, etc.)
+
+Nomad has a useful [tutorial](https://developer.hashicorp.com/nomad/tutorials/stateful-workloads/stateful-workloads-host-volumes) about how they work.
+
+More recent versions of Nomad have "dynamic host volumes" which would allow us to create these on the fly, not in Ansible, but we're not upgraded enough for that. Nomad has a [tutorial](https://developer.hashicorp.com/nomad/tutorials/stateful-workloads/stateful-workloads-dynamic-host-volumes) for those.
+
+### Application Configurations
+
+Containers deployed to the Nomad cluster may still need external systems provisioned and variables persisted to Nomad. For example an application may need a postgres database set up on our cluster, secrets installed into Nomad (which are currently in Ansible Vault), or other configuration done.
+
+Those things happen in [`tasks/application_configurations.yml`](tasks/application_configurations.yml).
+
+This pattern may change in the future. If you want to run a playbook without running these application configurations you can do: `ansible-playbook playbooks/nomad.yml --skip-tags app_config`. You may want to do that if you're just hoping to upgrade a client or server install, change the Nomad/Consul configuration, or other tasks that are about the cluster and not the applications on it.
+
+Similarly, to update a single app's configuration you can do something like `ansible-playbook playbooks/nomad.yml --tags dpulc`
```

Comment:
```suggestion
Similarly, to update a single app's configuration you can do something like `ansible-playbook playbooks/nomad.yml --tags dpulc`, because the include tasks all have tags on them.
```